From 863dd4efb2ed7c845b202bd28f5045ae39090253 Mon Sep 17 00:00:00 2001
From: Andrew Tridgell <tridge@samba.org>
Date: Fri, 18 May 2012 21:01:33 +1000
Subject: [PATCH 1/2] fixed configure without SDL

---
 libdc1394/configure.in |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/libdc1394/configure.in b/libdc1394/configure.in
index 5b82d8c..891ff2b 100644
--- a/libdc1394/configure.in
+++ b/libdc1394/configure.in
@@ -42,11 +42,11 @@ PKG_CHECK_MODULES(LIBUSB, [libusb-1.0],
     [AC_DEFINE(HAVE_LIBUSB,[],[Defined if libusb is present])],
     [AC_MSG_WARN([libusb-1.0 not found])])
 
-MIN_SDL_VERSION=1.2.4
-AH_TEMPLATE(HAVE_LIBSDL, [SDL library])
-AM_PATH_SDL($MIN_SDL_VERSION, [AC_DEFINE(HAVE_LIBSDL) HAVE_LIBSDL="yes"])
-AC_SUBST(SDL_CFLAGS)
-AC_SUBST(SDL_LIBS)
+#MIN_SDL_VERSION=1.2.4
+#AH_TEMPLATE(HAVE_LIBSDL, [SDL library])
+#AM_PATH_SDL($MIN_SDL_VERSION, [AC_DEFINE(HAVE_LIBSDL) HAVE_LIBSDL="yes"])
+#AC_SUBST(SDL_CFLAGS)
+#AC_SUBST(SDL_LIBS)
 
 case "$host" in
 *-*-linux*)
-- 
1.7.9.5

