From 5f8b7c39d4b90822c988f42ee55b9ec563ad081e Mon Sep 17 00:00:00 2001
From: Andrew Tridgell <tridge@samba.org>
Date: Fri, 18 May 2012 10:21:49 +1000
Subject: [PATCH 3/3] FTDI: added no_dtr option

---
 drivers/usb/serial/ftdi_sio.c |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/usb/serial/ftdi_sio.c b/drivers/usb/serial/ftdi_sio.c
index 60d7b1eb..1d3602b 100644
--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -59,6 +59,14 @@ static int debug;
 static __u16 vendor = FTDI_VID;
 static __u16 product;
 
+/* 
+ * when no_dtr is set, attempts to set DTR are ignored. This is used
+ * to prevent reset of DTR sensitive devices on reconnect (especially
+ * APM2 AutoPilots where DTR being asserted can cause a UAV to crash)
+ */
+static bool no_dtr;
+
+
 struct ftdi_private {
 	struct kref kref;
 	enum ftdi_chip_type chip_type;
@@ -1050,6 +1058,12 @@ static int update_mctrl(struct usb_serial_port *port, unsigned int set,
 
 	clear &= ~set;	/* 'set' takes precedence over 'clear' */
 	urb_value = 0;
+
+	if (no_dtr) {
+		clear &= ~TIOCM_DTR;
+		set   &= ~TIOCM_DTR;
+	}
+
 	if (clear & TIOCM_DTR)
 		urb_value |= FTDI_SIO_SET_DTR_LOW;
 	if (clear & TIOCM_RTS)
@@ -2439,4 +2453,5 @@ module_param(product, ushort, 0);
 MODULE_PARM_DESC(product, "User specified product ID");
 
 module_param(ndi_latency_timer, int, S_IRUGO | S_IWUSR);
+module_param(no_dtr, bool, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(ndi_latency_timer, "NDI device latency timer override");
-- 
1.7.9.5

